src/main.cpp

```diff
@@ -1140,8 +1140,17 @@ void vcRenderSceneUI(vcState *pProgramState, const ImVec2 &windowPos, const ImVe
         ImGui::InputScalarN(vcString::Get("sceneCameraRotation"), ImGuiDataType_Double, &pProgramState->camera.eulerRotation.x, 3);
         pProgramState->camera.eulerRotation = UD_DEG2RAD(pProgramState->camera.eulerRotation);
 
-        if (ImGui::SliderFloat(vcString::Get("sceneCameraMoveSpeed"), &(pProgramState->settings.camera.moveSpeed), vcSL_CameraMinMoveSpeed, vcSL_CameraMaxMoveSpeed, "%.3f m/s", 4.f))
-          pProgramState->settings.camera.moveSpeed = udClamp(pProgramState->settings.camera.moveSpeed, vcSL_CameraMinMoveSpeed, vcSL_GlobalLimitSmallf);
+        if (pProgramState->cameraInput.pAttachedToSceneItem != nullptr)
+        {
+          pProgramState->settings.camera.moveSpeed = udMin(pProgramState->settings.camera.moveSpeed, vcSL_CameraMaxAttachMoveSpeed);
```

Comment:
This isn't required, doing it after the slider changes is acceptable.

Reply:
I removed the branch, and reduced the camera's maximum movement speed to 500.